ui/src/app/components/direct-connection/direct-connection.component.spec.ts

```diff
@@ -0,0 +1,124 @@
+
+describe('DirectConnectionComponent', () => {
+  let component: DirectConnectionComponent
+  let fixture: ComponentFixture<DirectConnectionComponent>
+  let btn: any
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      declarations: [DirectConnectionComponent],
+      imports: [
+        RouterModule.forRoot(appRoutes),
+        HttpClientModule,
+        ReactiveFormsModule,
+        MatCardModule,
+        MatSelectModule,
+        MatOptionModule,
+        MatFormFieldModule,
+        MatInputModule,
+        BrowserAnimationsModule,
+        MatSnackBarModule,
+      ],
+      providers: [SnackbarService],
+    }).compileComponents()
+  })
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(DirectConnectionComponent)
+    component = fixture.componentInstance
+    fixture.detectChanges()
+    btn = fixture.debugElement.query(By.css('button[type=submit]'))
+  })
+
+  it('should create', () => {
+    expect(component).toBeTruthy()
+  })
+
+  it('should validate required input', () => {
+    let engine = component.connectForm.get('dbEngine')
+    engine?.setValue('')
+
+    expect(component.connectForm.valid).toBeFalsy()
+
+    expect(engine?.invalid).toBeTruthy()
+    engine?.setValue('mysql')
+    expect(engine?.invalid).toBeFalsy()
+
+    let host = component.connectForm.get('hostName')
+    host?.setValue('')
+    expect(host?.invalid).toBeTruthy()
+    host?.setValue('104.45.67.00')
+    expect(host?.invalid).toBeFalsy()
+
+    let port = component.connectForm.get('port')
+    port?.setValue('')
+    expect(port?.invalid).toBeTruthy()
+    port?.setValue('1433')
+    expect(port?.invalid).toBeFalsy()
+
+    let user = component.connectForm.get('userName')
+    user?.setValue('')
+    expect(user?.invalid).toBeTruthy()
+    user?.setValue('sa')
+    expect(user?.invalid).toBeFalsy()
+
+    let pass = component.connectForm.get('password')
+    pass?.setValue('')
+    expect(pass?.invalid).toBeTruthy()
+    pass?.setValue('23143')
+    expect(pass?.invalid).toBeFalsy()
+
+    let dbname = component.connectForm.get('dbName')
+    dbname?.setValue('')
+    expect(dbname?.invalid).toBeTruthy()
+    dbname?.setValue('mysql')
+    expect(dbname?.invalid).toBeFalsy()
+
+    // now every input is valid do form will be valid
+    expect(component.connectForm.valid).toBeTruthy()
+  })
+
+  it('should button disabled when form is invalid', () => {
+    let host = component.connectForm.get('hostName')
+    host?.setValue('')
+
+    expect(component.connectForm.invalid).toBeTruthy()
+    fixture.detectChanges()
+
+    expect(btn.nativeElement.disabled).toEqual(true)
+  })
+
+  it('should button enabled when form is valid', () => {
+    let host = component.connectForm.get('hostName')
+    host?.setValue('')
+    expect(component.connectForm.invalid).toBeTruthy()
+
+    component.connectForm.patchValue({
+      hostName: '123.345.678.56',
+      dbEngine: 'mysql',
+      port: '1433',
+      userName: 'sa',
```

Comment:
where are these values used ? We should clean it up and put sane values

Reply:
updated